src/core/utils/RbConstants.h

```diff
@@ -30,8 +33,40 @@ namespace RbConstants {
         const double    min         = std::numeric_limits<double>::min();
         const double    nan         = std::numeric_limits<double>::quiet_NaN();
         const double    neginf      = -std::numeric_limits<double>::infinity();
+
+	struct neginfWithReason
+	{
+	    template <typename T>
+	    neginfWithReason& operator<<(const T&);
+
+	    operator double() const {return neginf;}
+	};
+
+	template <typename T>
+	neginfWithReason& neginfWithReason::operator<<(const T& t)
+	{
+	    if (RbSettings::userSettings().getLogMCMC() >= 2)
+		std::cerr<<t;
+	    return *this;
+	}
+
+	struct nanWithReason
+	{
+	    template <typename T>
+	    nanWithReason& operator<<(const T&);
+
+	    operator double() const {return nan;}
```

Comment:
Just glancing over this on my tablet so might've missed something, but could you append the \n here (and above) instead of to each message to defeat the annoyance?

Reply:
Good question.  I think if you do `nanWithReasion()<<a<<b<<c` that will put a `\n` after `a`, `b`, and `c`, (three `\n`s) instead of once at the end.

I'm thinking I might store a `std::ostringstream` inside the `struct` instead.  Then we can add a '\n' when we actually print the message.